pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -62,8 +80,18 @@ func (c *CSIMaxVolumeLimitChecker) attachableLimitPredicate(
 		return true, nil, nil
 	}
 
+	// If either Node or CSINode objects are invalid, don't schedule pod on node
+	if nodeInfo == nil || nodeInfo.Node() == nil {
+		return false, nil, fmt.Errorf("invalid node")
+	}
+
+	csiNode, err := c.csiNodeInfo.GetCSINodeInfo(nodeInfo.Node().Name)
+	if err != nil {
+		return false, nil, err
+	}
+
```

Comment:
Is it fair to always required both `Node` and `CSINode`? The latter was introduced in v1.12.